src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -33,12 +38,22 @@ internal class CompletionHandler : RazorLSPHandlerBase, IRequestHandler<Completi
                 return null;
             }
 
+            // Switch to a background thread.
+            await TaskScheduler.Default;
+
             var projectionResult = await GetProjectionAsync(documentSnapshot, request.Position, cancellationToken);
             if (projectionResult == null)
             {
                 return null;
             }
 
+            if (request.Context.TriggerKind == CompletionTriggerKind.TriggerCharacter &&
+                !IsApplicableTriggerCharacter(request.Context.TriggerCharacter, projectionResult.LanguageKind))
+            {
+                // We were triggered but the trigger character doesn't make sense for the current cursor position. Bail.
```

Comment:
Is this the "Should log warning/debug" kind of "doesn't make sense"?

Reply:
No we expect this to happen very commonly like when someone enters `<` in C# context. We should just ignore those requests.